src/grt/src/RepairAntennas.cpp

```diff
@@ -855,4 +857,623 @@ double RepairAntennas::diffArea(odb::dbMTerm* mterm)
+  if (info.pin_num_near_to_start_ != 0 && info.pin_num_near_to_end_ != 0) {
+    req_tiles = std::max(2, int(req_tiles * 0.15));
+  }
+  int req_size = req_tiles * tile_size;
+  const int bridge_size = 2 * tile_size;
+  const int layer_level = violation_layer->getRoutingLevel();
+
+  int jumper_count = 0;
+  // place bridge in segment begin
+  if (info.pin_num_near_to_end_ == 0
+      || (info.pin_num_near_to_start_ != 0 && info.pin_num_near_to_end_ != 0)) {
+    // Get segment pos in route to add jumper
+    const int seg_pos = getSegmentIdToAdd(segment_ids,
+                                          route,
+                                          req_size,
+                                          bridge_size,
+                                          tile_size,
+                                          is_horizontal,
+                                          true);
+    // if has valid segment to add jumper
+    if (seg_pos != -1) {
+      auto& seg = route[seg_pos];
+      if (is_horizontal) {
+        // get jumper position
+        const int bridge_init_x = seg.init_x + req_size;
+        const int bridge_final_x = bridge_init_x + bridge_size;
+
+        jumper_count++;
+        addJumperHorizontal(
+            seg_pos, route, bridge_init_x, bridge_final_x, layer_level);
+      } else {
+        // Get jumper position
+        const int bridge_init_y = seg.init_y + req_size;
+        const int bridge_final_y = bridge_init_y + bridge_size;
+
+        jumper_count++;
+        addJumperVertical(
+            seg_pos, route, bridge_init_y, bridge_final_y, layer_level);
+      }
+    }
+  }
+  // if need place other in segment end
+  if (info.pin_num_near_to_start_ == 0
+      || (info.pin_num_near_to_start_ != 0 && info.pin_num_near_to_end_ != 0)) {
+    // Get segment pos in route to add jumper
+    const int seg_pos = getSegmentIdToAdd(segment_ids,
+                                          route,
+                                          req_size,
+                                          bridge_size,
+                                          tile_size,
+                                          is_horizontal,
+                                          false);
+    // if has valid segment to add jumper
+    if (seg_pos != -1) {
+      auto& seg = route[seg_pos];
+      if (is_horizontal) {
+        // Get jumper position
+        const int bridge_init_x = seg.final_x - req_size - bridge_size;
+        const int bridge_final_x = bridge_init_x + bridge_size;
+
+        jumper_count++;
+        addJumperHorizontal(
+            seg_pos, route, bridge_init_x, bridge_final_x, layer_level);
+      } else {
+        // Get jumper position
+        const int bridge_init_y = seg.final_y - req_size - bridge_size;
+        const int bridge_final_y = bridge_init_y + bridge_size;
+
+        jumper_count++;
+        addJumperVertical(
+            seg_pos, route, bridge_init_y, bridge_final_y, layer_level);
+      }
+    }
```

Comment:
This block of code is almost identical to the block above. Could you refactor it into a single function?